drivers/frequency/hmc7044/hmc7044.c

```diff
@@ -640,6 +645,122 @@ static int32_t hmc7044_setup(struct hmc7044_dev *dev)
 	return SUCCESS;
 }
 
+/**
+ * Setup the device.
+ * @param dev - The device structure.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+static int32_t hmc7043_setup(struct hmc7044_dev *dev)
+{
+	struct hmc7044_chan_spec *chan;
+	uint32_t i;
+
+	if (dev->clkin_freq_ccf[0])
+		dev->pll2_freq = dev->clkin_freq_ccf[0];
+	else
+		dev->pll2_freq  = dev->clkin_freq[0];
+
+	if (!dev->pll2_freq) {
+		printf("%s: Failed to get valid parent rate\n", __func__);
+		return FAILURE;
+	}
+
+	/* Resets all registers to default values */
+	hmc7044_write(dev, HMC7044_REG_SOFT_RESET, HMC7044_SOFT_RESET);
+	mdelay(10);
+	hmc7044_write(dev, HMC7044_REG_SOFT_RESET, 0);
+	mdelay(10);
+
+	/* Load the configuration updates (provided by Analog Devices) */
+	hmc7044_write(dev, HMC7044_REG_CLK_OUT_DRV_LOW_PW, 0x4d);
+	hmc7044_write(dev, HMC7044_REG_CLK_OUT_DRV_HIGH_PW, 0xdf);
+
+	/* Disable all channels */
+	for (i = 0; i < HMC7044_NUM_CHAN; i++)
+		hmc7044_write(dev, HMC7044_REG_CH_OUT_CRTL_0(i), 0);
+
+	if (dev->pll2_freq < 1000000000U)
+		hmc7044_write(dev, HMC7044_CLK_INPUT_CTRL,
+			      HMC7044_LOW_FREQ_INPUT_MODE);
+
+	hmc7044_write(dev, HMC7044_REG_EN_CTRL_0,
+		      (dev->rf_reseeder_en ? HMC7044_RF_RESEEDER_EN : 0) |
+		      HMC7044_SYSREF_TIMER_EN);
+
+	/* Program the SYSREF timer */
```

Comment:
nitpick: is this comment from this driver? I feel like it's leftover from porting.